draft-ietf-acme-acme.md

```diff
@@ -982,65 +982,73 @@ A client may wish to change the public key that is associated with a
 registration in order to recover from a key compromise or proactively mitigate
 the impact of an unnoticed key compromise.
 
-To change the key associated with an account, the client sends a POST request
-containing a key-change object with the following fields:
+To change the key associate with an account, the client first constructs a
+key-change object describing the change that it would like the server to make:
 
 oldKey (required, JWK):
-: The JWK representation of the original key (i.e., the client's current account
+: The JWK thumbprint of the original key (i.e., the client's current account
 key)
 
 newKey (requrired, JWK):
-: The JWK representation of the new key
+: The JWK thumbprint of the new key
 
-The JWS of this POST must have two signatures: one signature from the existing
-key on the account, and one signature from the new key that the client proposes
-to use. This demonstrates that the client actually has control of the
-private key corresponding to the new public key. The protected header must
-contain a JWK field containing the current account key.
+The client then encapsulates the key-change object in a JWS, signed with the
+client's current account key (i.e., the key matching the "oldKey" value).  This
+JWS is NOT REQUIRED to have the "nonce" and "url" fields, despite the general
+requirement that these fields be present.
```

Comment:
Good catch, @ekr.  It's actually very important to include the url in the inner JWS.  Otherwise, the holder of the new key has the ability to take over accounts using the old key on any ACME server, by replaying the inner JWS.  If the same person controls both keys, that's not a big deal, but if you're transferring your ACME account to a different person, it's quite important to prevent this.

It's also important, albeit less so, to include the nonce in the inner JWS.  Otherwise if you create a new account using the old key, the holder of the new key can take it over.  That's only a problem if you transfer your account to a different person and then reuse your old key for a new account.